treeherder/log_parser/artifactbuildercollection.pyx

```diff
@@ -101,30 +101,46 @@ BuildbotPerformanceDataArtifactBuilder
 
         Stream lines from the gzip file and run each parser against it,
         building the ``artifact`` as we go.
-
         """
+        with closing(self.get_log_handle(self.url)) as lh:
+
+            def _parse_lines(f, readpos, end=False):
+                f.seek(readpos)
+                for line in f:
+                    if '\n' not in line and not end:
```

Comment:
In case we stopped reading in a middle of a line, and we need more data to continue.